aspnetcore-webapi/UsingResultPatternInNETWebAPI/UsingResultPatternInNETWebAPI/TheResultPattern/CustomError.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace UsingResultPatternInNETWebAPI.TheResultPattern;
+
+public sealed record CustomError(string Code, string Message)
+{
+    private static readonly string RecordNotFoundCode = "RecordNotFound";
+    private static readonly string ValidationErrorCode = "ValidationError";
+
+    public static readonly CustomError None = new(string.Empty, string.Empty);
+
+    public static CustomError RecordNotFound(string message)
+    {
+        return new CustomError(RecordNotFoundCode, message);
+    }
+
+    public static CustomError ValidationError(string email)
+    {
```

Comment:
The ValidationError implementation should be same as RecordNotFound which will accept the message from caller. The current implementation cannot be used for other validation errors.